core/src/tx/mod.rs

```diff
@@ -168,7 +170,8 @@ impl<CallData: EncodeAsFields> TxPayload for Payload<CallData> {
             .map(|f| scale_encode::Field::new(f.ty.id, f.name.as_deref()));
 
         self.call_data
-            .encode_as_fields_to(&mut fields, metadata.types(), out)?;
+            .encode_as_fields_to(&mut fields, metadata.types(), out)
+            .expect("The fields are valid types from the metadata, qed;");
```

Comment:
There are various other reasons why the call data might not properly encode though; better to keep the `?`?